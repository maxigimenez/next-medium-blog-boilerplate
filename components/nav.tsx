import React from 'react';
import Link from 'next/link';
import config from '@/config';
import { Category } from '@/core/models';

type Props = {
  categories: Category[];
};

export const Nav = ({ categories }: Props) => (
  <nav className="topnav navbar navbar-expand-lg navbar-light bg-white fixed-top">
    <div className="container">
      <Link href="/">
        <a href="/" className="navbar-brand">
          <strong>{config.name}</strong>
        </a>
      </Link>
      <button
        className="navbar-toggler collapsed"
        type="button"
        data-toggle="collapse"
        data-target="#navbarColor02"
        aria-controls="navbarColor02"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="navbar-collapse collapse" id="navbarColor02">
        <ul className="navbar-nav mr-auto d-flex align-items-center">
          {categories &&
            categories.map((category: Category) => {
              return (
                <li key={category.slug} className="nav-item">
                  <Link href={`/category/${category.slug}`}>
                    <a href={`/category/${category.slug}`} className="nav-link">
                      {category.name}
                    </a>
                  </Link>
                </li>
              );
            })}
        </ul>
        <ul className="navbar-nav ml-auto d-flex align-items-center">
          <li className="nav-item highlight">
            <a
              className="nav-link"
              href="https://github.com/maxigimenez/next-medium-blog-boilerplate"
              target="_blank"
              rel="noreferrer"
            >
              Get this blog boilerplate
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);
