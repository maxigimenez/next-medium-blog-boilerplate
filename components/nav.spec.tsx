import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './nav';

describe('nav', () => {
  describe('snapshots', () => {
    test('render without categories', () => {
      const component = shallow(<Nav categories={[]} />);
      expect(component).toMatchSnapshot();
    });

    test('render with categories', () => {
      const categories = [
        {
          name: 'Category 1',
          slug: 'category-1',
        },
        {
          name: 'Category 2',
          slug: 'category-2',
        },
      ];
      const component = shallow(<Nav categories={categories} />);
      expect(component).toMatchSnapshot();
    });

    test('render if categories undefined', () => {
      const component = shallow(<Nav categories={undefined} />);
      expect(component).toMatchSnapshot();
    });
  });
});
