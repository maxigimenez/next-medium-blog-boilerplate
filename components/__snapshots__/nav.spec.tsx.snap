// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav snapshots render if categories undefined 1`] = `
<nav
  className="topnav navbar navbar-expand-lg navbar-light bg-white fixed-top"
>
  <div
    className="container"
  >
    <Link
      href="/"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <strong>
          Demo Blog
        </strong>
      </a>
    </Link>
    <button
      aria-controls="navbarColor02"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      data-target="#navbarColor02"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      id="navbarColor02"
    >
      <ul
        className="navbar-nav mr-auto d-flex align-items-center"
      />
      <ul
        className="navbar-nav ml-auto d-flex align-items-center"
      >
        <li
          className="nav-item highlight"
        >
          <a
            className="nav-link"
            href="https://github.com/maxigimenez/next-medium-blog-boilerplate"
            rel="noreferrer"
            target="_blank"
          >
            Get this blog boilerplate
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`nav snapshots render with categories 1`] = `
<nav
  className="topnav navbar navbar-expand-lg navbar-light bg-white fixed-top"
>
  <div
    className="container"
  >
    <Link
      href="/"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <strong>
          Demo Blog
        </strong>
      </a>
    </Link>
    <button
      aria-controls="navbarColor02"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      data-target="#navbarColor02"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      id="navbarColor02"
    >
      <ul
        className="navbar-nav mr-auto d-flex align-items-center"
      >
        <li
          className="nav-item"
          key="category-1"
        >
          <Link
            href="/category/category-1"
          >
            <a
              className="nav-link"
              href="/category/category-1"
            >
              Category 1
            </a>
          </Link>
        </li>
        <li
          className="nav-item"
          key="category-2"
        >
          <Link
            href="/category/category-2"
          >
            <a
              className="nav-link"
              href="/category/category-2"
            >
              Category 2
            </a>
          </Link>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto d-flex align-items-center"
      >
        <li
          className="nav-item highlight"
        >
          <a
            className="nav-link"
            href="https://github.com/maxigimenez/next-medium-blog-boilerplate"
            rel="noreferrer"
            target="_blank"
          >
            Get this blog boilerplate
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`nav snapshots render without categories 1`] = `
<nav
  className="topnav navbar navbar-expand-lg navbar-light bg-white fixed-top"
>
  <div
    className="container"
  >
    <Link
      href="/"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <strong>
          Demo Blog
        </strong>
      </a>
    </Link>
    <button
      aria-controls="navbarColor02"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      data-target="#navbarColor02"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="navbar-collapse collapse"
      id="navbarColor02"
    >
      <ul
        className="navbar-nav mr-auto d-flex align-items-center"
      />
      <ul
        className="navbar-nav ml-auto d-flex align-items-center"
      >
        <li
          className="nav-item highlight"
        >
          <a
            className="nav-link"
            href="https://github.com/maxigimenez/next-medium-blog-boilerplate"
            rel="noreferrer"
            target="_blank"
          >
            Get this blog boilerplate
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
